src/code.cloudfoundry.org/garden-external-networker/manager/manager.go

```diff
@@ -111,7 +112,7 @@ func (m *Manager) Up(containerHandle string, inputs UpInputs) (*UpOutputs, error
 		return nil, errors.New("cni up failed: no ip allocated")
 	}
 
-	result020, err := result.GetAsVersion("0.2.0")
+	result040, err := result.GetAsVersion("0.4.0")
```

Comment:
I'm a little concerned with this bump and impact it might have to CNIs other than silk. But I also thought we bumped CNI versions to 1.0.0 recently, so not sure if this was something that was missed, or what happened. I'll dig more into this though - no action needed on your part currently.

Reply:
> I'm a little concerned with this bump and impact it might have to CNIs other than silk. But I also thought we bumped CNI versions to 1.0.0 recently, so not sure if this was something that was missed, or what happened. I'll dig more into this though - no action needed on your part currently.

Great questions! That's also one point where we expected some questions and would be glad to get feedback.

[We found this here in the silk-release](https://github.com/cloudfoundry/silk-release/blob/207ae9320ed4d01c5a0b839dfb6bd99e213a2ae9/src/code.cloudfoundry.org/cni-wrapper-plugin/main.go#L422-L426), that configured the supported CNI version to `1.0.0`, that's why we didn't add any backwards compatibility and always expect an interface name to be returned.

Should we add backwards compatibility anyway? One of the main reasons why we didn't do this is that this metric then becomes "optional". We would also need to change the implementation in guardian to allow an empty/no interface name to be returned and just skip reading the network metrics in that case. Currently it's a required return value.

But we might anyway have to be more relaxed about this requirement if I [interpret your comment here correctly](https://github.com/cloudfoundry/cf-networking-release/issues/64#issuecomment-1662360577) regarding backwards compatibility between the different CF components.

Reply:
Hi @geofffranks,

as discussed we implemented backwards compatibility for CNI versions < 0.4.0 in aa9963204ad387d0f1859848caf8cb06f43d5d6d.

The interface name is now an optional property that will only be returned for CNI versions > 0.4.0.